applications/ContactStructuralMechanicsApplication/custom_conditions/ALM_frictionless_mortar_contact_condition.h

```diff
@@ -265,14 +303,9 @@ class AugmentedLagrangianMethodFrictionlessMortarContactCondition: public Augmen
     unsigned int GetActiveInactiveValue(GeometryType& CurrentGeometry) const override
     {
         unsigned int value = 0;
-        
-        for (unsigned int i_node = 0; i_node < CurrentGeometry.size(); i_node++)
-        {
-            if ((CurrentGeometry[i_node].Is(ACTIVE) == true) || (this->Is(VISITED) == true))
-            {
+        for (unsigned int i_node = 0; i_node < TNumNodes; ++i_node)
+            if (CurrentGeometry[i_node].Is(ACTIVE) == true)
                 value += std::pow(2, i_node);
```

Comment:
are u sure of this? you are making 2 to the power i_node

Reply:
if it is what you want to do then use "bit shifting" it will be way faster than pow

Reply:
Do you mean this?

https://stackoverflow.com/questions/141525/what-are-bitwise-shift-bit-shift-operators-and-how-do-they-work